src/frequenz/dispatch/_bg_service.py

```diff
@@ -23,6 +27,77 @@
 """The logger for this module."""
 
 
+class _MergeStrategy(ABC):
+    """Base class for strategies to merge running intervals."""
+
+    @abstractmethod
+    def _get_filter_function(
+        self,
+        scheduler: DispatchScheduler,
+    ) -> Callable[[Dispatch], bool]:
+        """Get a filter function for dispatches.
+
+        Args:
+            scheduler: The dispatch scheduler.
+
+        Returns:
+            A filter function.
+        """
+
+
+class MergeByType(_MergeStrategy):
+    """Merge running intervals based on the dispatch type."""
+
+    def __init__(self) -> None:
+        """Initialize the strategy."""
+        self._scheduler: DispatchScheduler
+        self._new_dispatch: Dispatch
+
+    def _get_filter_function(
+        self, scheduler: DispatchScheduler
+    ) -> Callable[[Dispatch], bool]:
+        """Get a filter function for dispatches."""
+        self._scheduler = scheduler
+        return self._filter_func
+
+    def _criteria(self, dispatch: Dispatch) -> bool:
+        """Define the criteria for checking other running dispatches."""
+        return dispatch.type == self._new_dispatch.type
+
+    def _filter_func(self, new_dispatch: Dispatch) -> bool:
+        """Filter dispatches based on the merge strategy.
+
+        Keeps start events.
+        Keeps stop events only if no other dispatches matching the
+        strategy's criteria are running.
+        """
+        if new_dispatch.started:
+            return True
+
+        self._new_dispatch = new_dispatch
+
+        # pylint: disable=protected-access
+        other_dispatches_running = any(
+            dispatch.started
+            for dispatch in self._scheduler._dispatches.values()
```

Comment:
`_dispatches` needs to be public too. Maybe it is best to expose the cached dispatches in the `Dispatcher` directly, as a read-only `collections.abc.Map`.